lib/ui/window.dart

```diff
@@ -730,54 +755,130 @@ class Window {
+  WindowPadding/*!*/ get padding => viewConfiguration.padding;
 
-  /// A callback that is invoked whenever the [devicePixelRatio],
-  /// [physicalSize], [padding], [viewInsets], or [systemGestureInsets]
-  /// values change, for example when the device is rotated or when the
-  /// application is resized (e.g. when showing applications side-by-side
-  /// on Android).
+  /// Updates the view's rendering on the GPU with the newly provided
+  /// [Scene].
+  ///
+  /// This function must be called within the scope of the
+  /// [PlatformDispatcher.onBeginFrame] or [PlatformDispatcher.onDrawFrame]
+  /// callbacks being invoked. If this function is called a second time during a
+  /// single [PlatformDispatcher.onBeginFrame]/[PlatformDispatcher.onDrawFrame]
+  /// callback sequence or called outside the scope of those callbacks, the call
+  /// will be ignored.
   ///
-  /// The engine invokes this callback in the same zone in which the callback
-  /// was set.
+  /// To record graphical operations, first create a [PictureRecorder], then
+  /// construct a [Canvas], passing that [PictureRecorder] to its constructor.
+  /// After issuing all the graphical operations, call the
+  /// [PictureRecorder.endRecording] function on the [PictureRecorder] to obtain
+  /// the final [Picture] that represents the issued graphical operations.
   ///
-  /// The framework registers with this callback and updates the layout
-  /// appropriately.
+  /// Next, create a [SceneBuilder], and add the [Picture] to it using
+  /// [SceneBuilder.addPicture]. With the [SceneBuilder.build] method you can
+  /// then obtain a [Scene] object, which you can display to the user via this
+  /// [render] function.
   ///
   /// See also:
   ///
-  ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
-  ///    register for notifications when this is called.
-  ///  * [MediaQuery.of], a simpler mechanism for the same.
-  VoidCallback/*?*/ get onMetricsChanged => _onMetricsChanged;
-  VoidCallback/*?*/ _onMetricsChanged;
-  Zone/*!*/ _onMetricsChangedZone = Zone.root;
+  ///  * [SchedulerBinding], the Flutter framework class which manages the
+  ///    scheduling of frames.
+  ///  * [RendererBinding], the Flutter framework class which manages layout and
+  ///    painting.
+  void render(Scene/*!*/ scene) => platformDispatcher.render(scene, this);
+}
+
+/// A top-level platform window displaying a Flutter layer tree drawn from a
+/// [Scene].
+///
+/// The current list of all Flutter views for the application is available from
+/// `WidgetsBinding.instance.platformDispatcher.views`. Only views that are of
+/// type [FlutterWindow] are top level platform windows.
+///
+/// There is also a [PlatformDispatcher.instance] singleton object in `dart:ui`
+/// if `WidgetsBinding` is unavailable, but we strongly advise avoiding a static
+/// reference to it. See the documentation for [PlatformDispatcher.instance] for
+/// more details about why it should be avoided.
```

Comment:
general tone note for all these docs (please go through all the docs affected by this PR and apply this throughout): the flutter voice avoids asking developers to do things or advising that developers do things. Instead, we tell people to "consider" doing things and we state the benefits and risks of each approach.